### ISSUE TITLE
segfault on switching with hotkeys

### ISSUE THREAD

**Issue by sriram109** [zevweiss/enthrall#2]
master: osx mavericks
remote: linux rhel6
when i attempt to move from the master to remote by sliding, it seems to just hang; i can switch back to the master (via hotkey), but upon switching with hotkey back to remote, enthrall crashes.

log below
SJCE413400:enthrall 25152$ ./enthrall.macosx-10.9 example.conf
[58977] 2015-01-30 13:20:22: initiating connection attempt to remote foo...
[58977] 2015-01-30 13:20:25: foo: XRandR extension version 1.3
[58977] 2015-01-30 13:20:25: foo: XInput extension version 2.0
[58977] 2015-01-30 13:20:25: foo: XTest extension version 2.2
[58977] 2015-01-30 13:20:25: remote foo becomes ready.
[58977] 2015-01-30 13:20:25: foo screen dimensions: 1919x1079
[58977] 2015-01-30 13:20:25: setting foo brightness to 1
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.975
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.95
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.925
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.9
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.875
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.85
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.825
[58977] 2015-01-30 13:20:25: setting foo brightness to 0.8
[58977] 2015-01-30 13:20:28: setting foo brightness to 0.8
[58977] 2015-01-30 13:20:28: Warning: CGEvent type 0 unknown
[58977] 2015-01-30 13:20:28: Warning: CGEvent type 29 unknown
[58977] 2015-01-30 13:20:28: setting foo brightness to 1
[58977] 2015-01-30 13:20:28: setting foo brightness to 0.8
[58977] 2015-01-30 13:20:28: setting foo brightness to 0.825
[58977] 2015-01-30 13:20:28: setting foo brightness to 0.975
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.825
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.85
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.95
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.85
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.875
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.925
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.875
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.9
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.9
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.9
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.925
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.875
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.925
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.95
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.85
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.95
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.975
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.825
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.975
[58977] 2015-01-30 13:20:29: setting foo brightness to 1
[58977] 2015-01-30 13:20:29: setting foo brightness to 0.8
[58977] 2015-01-30 13:20:29: setting foo brightness to 1
[58977] 2015-01-30 13:20:30: Warning: CGEvent type 14 unknown
[58977] 2015-01-30 13:20:40: setting foo brightness to 1
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.975
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.95
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.925
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.9
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.875
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.85
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.825
[58977] 2015-01-30 13:20:40: setting foo brightness to 0.8
[58977] 2015-01-30 13:20:47: setting foo brightness to 0.8
*** glibc detected *** /home/srsubramanian/enthrall/enthrall.rhel6: free(): invalid next size (normal): 0x0000000000685e20 ***
======= Backtrace: =========
/lib64/libc.so.6[0x307ee76126]
/lib64/libc.so.6[0x307ee78c53]
/lib64/libc.so.6(xdr_string+0xc8)[0x307ef17b28]
/home/srsubramanian/enthrall/enthrall.rhel6[0x40f148]
/home/srsubramanian/enthrall/enthrall.rhel6[0x40f34b]
/lib64/libc.so.6(xdr_free+0x15)[0x307ef172d5]
/home/srsubramanian/enthrall/enthrall.rhel6[0x407003]
/home/srsubramanian/enthrall/enthrall.rhel6[0x407405]
/home/srsubramanian/enthrall/enthrall.rhel6[0x40ad72]
/home/srsubramanian/enthrall/enthrall.rhel6[0x40ae63]
/home/srsubramanian/enthrall/enthrall.rhel6[0x406908]
/home/srsubramanian/enthrall/enthrall.rhel6[0x406364]
/lib64/libc.so.6(__libc_start_main+0xfd)[0x307ee1ecdd]
/home/srsubramanian/enthrall/enthrall.rhel6[0x403099]
======= Memory map: ========
00400000-00417000 r-xp 00000000 fd:02 1048664                            /home/srsubramanian/enthrall/enthrall.rhel6
00616000-00657000 rw-p 00016000 fd:02 1048664                            /home/srsubramanian/enthrall/enthrall.rhel6
0066a000-0068b000 rw-p 00000000 00:00 0                                  [heap]
307e600000-307e620000 r-xp 00000000 fd:00 1048604                        /lib64/ld-2.12.so
307e81f000-307e820000 r--p 0001f000 fd:00 1048604                        /lib64/ld-2.12.so
307e820000-307e821000 rw-p 00020000 fd:00 1048604                        /lib64/ld-2.12.so
307e821000-307e822000 rw-p 00000000 00:00 0 
307ea00000-307ea83000 r-xp 00000000 fd:00 1065235                        /lib64/libm-2.12.so
307ea83000-307ec82000 ---p 00083000 fd:00 1065235                        /lib64/libm-2.12.so
307ec82000-307ec83000 r--p 00082000 fd:00 1065235                        /lib64/libm-2.12.so
307ec83000-307ec84000 rw-p 00083000 fd:00 1065235                        /lib64/libm-2.12.so
307ee00000-307ef8a000 r-xp 00000000 fd:00 1048606                        /lib64/libc-2.12.so
307ef8a000-307f189000 ---p 0018a000 fd:00 1048606                        /lib64/libc-2.12.so
307f189000-307f18d000 r--p 00189000 fd:00 1048606                        /lib64/libc-2.12.so
307f18d000-307f18e000 rw-p 0018d000 fd:00 1048606                        /lib64/libc-2.12.so
307f18e000-307f193000 rw-p 00000000 00:00 0 
307f600000-307f617000 r-xp 00000000 fd:00 1048646                        /lib64/libpthread-2.12.so
307f617000-307f817000 ---p 00017000 fd:00 1048646                        /lib64/libpthread-2.12.so
307f817000-307f818000 r--p 00017000 fd:00 1048646                        /lib64/libpthread-2.12.so
307f818000-307f819000 rw-p 00018000 fd:00 1048646                        /lib64/libpthread-2.12.so
307f819000-307f81d000 rw-p 00000000 00:00 0 
307fa00000-307fa07000 r-xp 00000000 fd:00 1048663                        /lib64/librt-2.12.so
307fa07000-307fc06000 ---p 00007000 fd:00 1048663                        /lib64/librt-2.12.so
307fc06000-307fc07000 r--p 00006000 fd:00 1048663                        /lib64/librt-2.12.so
307fc07000-307fc08000 rw-p 00007000 fd:00 1048663                        /lib64/librt-2.12.so
3082200000-308221d000 r-xp 00000000 fd:00 669351                         /usr/lib64/libxcb.so.1.1.0
308221d000-308241d000 ---p 0001d000 fd:00 669351                         /usr/lib64/libxcb.so.1.1.0
308241d000-308241e000 rw-p 0001d000 fd:00 669351                         /usr/lib64/libxcb.so.1.1.0
3082a00000-3082a02000 r-xp 00000000 fd:00 669292                         /usr/lib64/libXau.so.6.0.0
3082a02000-3082c02000 ---p 00002000 fd:00 669292                         /usr/lib64/libXau.so.6.0.0
3082c02000-3082c03000 rw-p 00002000 fd:00 669292                         /usr/lib64/libXau.so.6.0.0
3087200000-3087216000 r-xp 00000000 fd:00 1053815                        /lib64/libgcc_s-4.4.7-20120601.so.1
3087216000-3087415000 ---p 00016000 fd:00 1053815                        /lib64/libgcc_s-4.4.7-20120601.so.1
3087415000-3087416000 rw-p 00015000 fd:00 1053815                        /lib64/libgcc_s-4.4.7-20120601.so.1
3577000000-3577012000 r-xp 00000000 fd:00 693239                         /usr/lib64/libXext.so.6.4.0
3577012000-3577212000 ---p 00012000 fd:00 693239                         /usr/lib64/libXext.so.6.4.0
3577212000-3577213000 rw-p 00012000 fd:00 693239                         /usr/lib64/libXext.so.6.4.0
3577800000-3577937000 r-xp 00000000 fd:00 693238                         /usr/lib64/libX11.so.6.3.0
3577937000-3577b37000 ---p 00137000 fd:00 693238                         /usr/lib64/libX11.so.6.3.0
3577b37000-3577b3d000 rw-p 00137000 fd:00 693238                         /usr/lib64/libX11.so.6.3.0
3577c00000-3577c02000 r-xp 00000000 fd:00 1067241                        /lib64/libdl-2.12.so
3577c02000-3577e02000 ---p 00002000 fd:00 1067241                        /lib64/libdl-2.12.so
3577e02000-3577e03000 r--p 00002000 fd:00 1067241                        /lib64/libdl-2.12.so
3577e03000-3577e04000 rw-p 00003000 fd:00 1067241                        /lib64/libdl-2.12.so
3f20000000-3f20009000 r-xp 00000000 fd:00 693040                         /usr/lib64/libXrender.so.1.3.0
3f20009000-3f20209000 ---p 00009000 fd:00 693040                         /usr/lib64/libXrender.so.1.3.0
3f20209000-3f2020a000 rw-p 00009000 fd:00 693040                         /usr/lib64/libXrender.so.1.3.0
3f20400000-3f2040e000 r-xp 00000000 fd:00 693249                         /usr/lib64/libXi.so.6.1.0
3f2040e000-3f2060e000 ---p 0000e000 fd:00 693249                         /usr/lib64/libXi.so.6.1.0
3f2060e000-3f2060f000 rw-p 0000e000 fd:00 693249                         /usr/lib64/libXi.so.6.1.0
3f20c00000-3f20c0a000 r-xp 00000000 fd:00 693250                         /usr/lib64/libXrandr.so.2.2.0
3f20c0a000-3f20e09000 ---p 0000a000 fd:00 693250                         /usr/lib64/libXrandr.so.2.2.0
3f20e09000-3f20e0a000 rw-p 00009000 fd:00 693250                         /usr/lib64/libXrandr.so.2.2.0
3f22400000-3f22405000 r-xp 00000000 fd:00 693251                         /usr/lib64/libXfixes.so.3.1.0
3f22405000-3f22605000 ---p 00005000 fd:00 693251                         /usr/lib64/libXfixes.so.3.1.0
3f22605000-3f22606000 rw-p 00005000 fd:00 693251                         /usr/lib64/libXfixes.so.3.1.0
3f23c00000-3f23c09000 r-xp 00000000 fd:00 693252                         /usr/lib64/libXcursor.so.1.0.2
3f23c09000-3f23e09000 ---p 00009000 fd:00 693252                         /usr/lib64/libXcursor.so.1.0.2
3f23e09000-3f23e0a000 rw-p 00009000 fd:00 693252                         /usr/lib64/libXcursor.so.1.0.2
3f26c00000-3f26c05000 r-xp 00000000 fd:00 693087                         /usr/lib64/libXtst.so.6.1.0
3f26c05000-3f26e05000 ---p 00005000 fd:00 693087                         /usr/lib64/libXtst.so.6.1.0
3f26e05000-3f26e06000 rw-p 00005000 fd:00 693087                         /usr/lib64/libXtst.so.6.1.0
7f94599de000-7f94599e5000 rw-p 00000000 00:00 0 
7f94599f9000-7f94599fb000 rw-p 00000000 00:00 0 
7fff7da64000-7fff7da85000 rw-p 00000000 00:00 0                          [stack]
7fff7db30000-7fff7db31000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
[58977] 2015-01-30 13:20:47: setting foo brightness to 0.825
[58977] 2015-01-30 13:20:47: Error: disconnecting remote 'foo': msgchan error
[58977] 2015-01-30 13:20:47: setting foo brightness to 0.85
Segmentation fault: 11

**Comment by sriram109** [comment]
steps to reproduce
1. start enthrall on osx
2. switch to remote (hotkey)
3. switch to master (hotkey)
4. switch to remote (hotkey) - crash

**Comment by zevweiss** [comment]
Unable to reproduce -- didn't crash, then retried with valgrind on the remote instance, didn't get any complaints from it.  Can you paste your config?

**Comment by sriram109** [comment]
# The master block sets global options and defaults for remotes.  The
# master node can be given an optional alias ("fred" here); in the
# topology block and hotkey focus actions, the master node can be
# referred to by either this alias (as a quoted string) or the
# (unquoted) keyword 'master'.
master "fred" {
	# log-file: if set to a quoted string, sets the path of a file
	# to which log messages are written.  Can alternately be set
	# to one of the keywords 'none' (disable logging), 'stderr'
	# (send log messages to stderr), or 'syslog' (send log
	# messages to the system log).
	#
	# log-file = "/home/me/enthrall.log"
	# log-file = syslog
	# log-file = stderr

	# log-level: sets the degree of verbosity in what messages are
	# sent to the log.  Must be one of the following keywords:
	# error, warn, info, verbose, debug.
	#
    log-level = debug

	# remote-shell: provides the path to the ssh binary to use.
	# If not provided, the default is simply "ssh" (whatever is
	# first in $PATH).
	#
	# remote-shell = "/alternate/path/to/ssh"

	# port: if provided, this sets the argument passed to ssh's
	# "-p" flag to set the default port to connect to on remotes.
	#
	# port = 2244

	# bind-address: if specified, the given string is passed as
	# the argument to ssh's "-b" flag.  May be useful if you have
	# multiple NICs, a v4 and a v6 address on the same interface,
	# etc.
	#
	# bind-address = "2607:f8b0:4009:803::1009"

	# identity-file: if provided, sets the argument passed to
	# ssh's "-i" flag to set the default key used to authenticate
	# to remotes.
	#
	# identity-file = "~/.ssh/enthrall-key"

	# user: if provided, this sets the default argument passed to
	# ssh's "-l" flag to set the username to connect as on the
	# remote system.
	#
	# user = "jschmoe"

	# remote-command: the default path at which to find the
	# enthrall binary on the remotes.  If not provided, the
	# default is simply whatever 'enthrall' is found first in
	# $PATH.
	#
	# remote-command = "/usr/local/bin/enthrall"

	# reconnect-max-tries: the maximum number of times to attempt
	# reconnecting to a failed remote before giving up.  Defaults
	# to 10.
	#
	# reconnect-max-tries = 5

	# reconnect-max-interval: the maximum number of seconds to
	# wait between reconnection attempts.  Default is 30.
	#
	# reconnect-max-interval = 10

	# show-focus: selects one of the following modes of providing
	# a visual hint of which node is focused (default is none):
	#
	#   dim-inactive BRIGHTNESS [DURATION NUMSTEPS]: the screens
	#   of non-active nodes are dimmed to the given fraction of
	#   their default brightness (~0.7 is probably a good starting
	#   point).  If DURATION and NUMSTEPS are specified, fade
	#   brightness transitions in NUMSTEPS steps over DURATION
	#   seconds (defaulting to 0 and 1, i.e. no fade).
	#
	#   flash-active BRIGHTNESS DURATION [NUMSTEPS]: when
	#   switching focus to a node, momentarily set its screen
	#   brightness to the given level (where "normal" is 1.0) for
	#   the specified time (in seconds).  If NUMSTEPS is
	#   specified, fade the flash out in the given number of steps
	#   over DURATION (defaulting to 1, i.e. no fade).  Note that
	#   nothing stops you from setting BRIGHTNESS to be less than
	#   1.0, so flash-active could also effectively act as a sort
	#   of "blink-active".
	#
	#   none: no visual focus hint.
	#
	# show-focus = dim-inactive 0.8 0.3 8
	# show-focus = flash-active 1.2 0.3 8
	# show-focus = flash-active 0.6 0.3 8

	# mouse-switch: selects one of the following mechanisms for
	# switching the focused node via the mouse (default is none):
	#
	#   slide: simply slide the mouse pointer off the edge of one
	#   node's display to have it slide onto the next node.
	#
	#   multi-tap COUNT WINDOW: "tap" the mouse pointer against an
	#   edge of a screen COUNT times within WINDOW seconds to
	#   switch focus in the corresponding direction.  With a COUNT
	#   of 1, WINDOW is ignored (this is equivalent to the 'slide'
	#   setting).  The maximum value for COUNT is determined by
	#   the macro EDGESTATE_HISTLEN if you really want to have to
	#   tap a lot of times to switch.
	#
	#   none: switch-by-mouse disabled.
	#
	mouse-switch = slide
	#mouse-switch = multi-tap 2 1

	# show-null-switch: selects one of the following settings for
	# determining whether or not to trigger the configured visual
	# focus-hint mechanism (see 'show-focus') when a "null" switch
	# is performed (e.g. switching left from a node that has no
	# left neighbor or switching to an already-focused node):
	#
	#   yes: do trigger visual focus hint.
	#
	#   no: do not trigger visual focus hint.
	#
	#   hotkey-only: trigger only if the null switch was performed
	#   via a hotkey, not if done via mouse-switch.
	#
	# show-null-switch: hotkey-only

	# hotkeys: hotkey[HOTKEY_STRING] = ACTION
	#
	# valid values for HOTKEY_STRING are system-dependent; some
	# possible examples on X11 might look like:
	#   "mod1+mod4+L"
	#   "mod1+mod4+control+Right"
	#
	# Similar things on OSX:
	#   "command+option+l"
	#   "command+option+control+RightArrow"
	#
	# ACTION can be one of the following:
	#
	#   focus DIRECTION (where DIRECTION is one of left, right,
	#   up, or down): switch focus to the given neighbor of the
	#   currently-focused node.
	#
	#   focus NODE: switch focus directly to the specified NODE,
	#   which can be either a string, naming a node by alias, or
	#   the keyword 'master' -- note that since master is a
	#   keyword (not a string) it should not be quoted.
	#
	#   focus previous: switch focus to the previously-focused
	#   node (toggle between the two most recently focused nodes).
	#
	#   reconnect: reset permanently-failed remotes to allow
	#   auto-reconnection to attempt to re-establish a connection.
	#
	#   quit: disconnect all remotes and terminate enthrall.

	# arrow-key-based direction-switching
	hotkey["command+option+control+LeftArrow"] = focus left
	hotkey["command+option+control+RightArrow"] = focus right
	hotkey["command+option+control+UpArrow"] = focus up
	hotkey["command+option+control+DownArrow"] = focus down

	# vi-style direction-switching
	hotkey["command+option+control+h"] = focus left
	hotkey["command+option+control+l"] = focus right
	hotkey["command+option+control+k"] = focus up
	hotkey["command+option+control+j"] = focus down

	# direct-switch keys
	hotkey["command+option+control+1"] = focus "foo"
	hotkey["command+option+control+2"] = focus master

	# focus toggle
	hotkey["command+option+control+Grave"] = focus previous

	hotkey["command+option+control+r"] = reconnect
}

# A remote block defines a remote node with a given alias.
remote "foo" {
	# hostname: if provided, sets the network hostname to connect
	# to for this remote.  If not provided, defaults to the
	# remote's alias (e.g. "foo" in this example remote block).
	#
	# hostname = "somehost"
	# hostname = "192.168.2.4"
	hostname = "10.11.189.22"

	# Each remote can also specify remote-shell, user, port,
	# bind-address, identity-file, and remote-command to provide a
	# per-remote override of the global defaults.
	#
	# remote-shell = "/alternate/path/to/ssh"
	# port = 22
	# identity-file = "~/.ssh/alternate-enthrall-key"
	# bind-address = "192.168.2.2"
	user = "srsubramanian"
	remote-command = "/home/srsubramanian/enthrall/enthrall.rhel6"

	# Platform-specific remote configuration parameters are
	# specified as strings in a special unstructured 'param' map.
	# Currently the only setting that's used for anything is
	# DISPLAY to set the environment variable of the same name on
	# X11 systems, so X11 remotes should probably have something
	# like:
	#
	param["DISPLAY"] = ":0.0"
}

# The topology block specifies how nodes are arranged relative to each
# other (i.e. who's on which side of whom).
topology {
	# Each link in the node topology is specified as:
	#
	#   NODE DIRECTION = NODE [DIRECTION]
	#
	# where NODEs are node aliases or the keyword master, as used
	# elsewhere, and DIRECTIONs are (again) one of left, right, up
	# and down.  For example:
	#
	#   master right = "foo" left
	#   "foo" right = "bar" left
	#   "fred" down = "baz" up
	#
	# This specifies a topology like this:
	#
	#    master <-> foo <-> bar
	#      ^
	#      |
	#      v
	#     baz
	#
	master right = "foo" left

	# Note, however, that the DIRECTION specifier on the right
	# side of the equal sign is optional; if omitted, this
	# specifies a one-directional link.  Example:
	#
	#   master up = "foo"
	#   "foo" down = "bar"
	#
	# If you so desire, you can use this to create exotic
	# topologies wherein (for example) a 'focus left' followed by
	# a 'focus right' does *not* leave you where you started.
	# Knock yourself out.
}

**Comment by zevweiss** [comment]
Thanks...for what it's worth I think if you surround it with a triple-backtick delimiter on its own line before & after it'll include it literally...

**Comment by sriram109** [comment]
```
# The master block sets global options and defaults for remotes.  The
# master node can be given an optional alias ("fred" here); in the
# topology block and hotkey focus actions, the master node can be
# referred to by either this alias (as a quoted string) or the
# (unquoted) keyword 'master'.
master "fred" {
	# log-file: if set to a quoted string, sets the path of a file
	# to which log messages are written.  Can alternately be set
	# to one of the keywords 'none' (disable logging), 'stderr'
	# (send log messages to stderr), or 'syslog' (send log
	# messages to the system log).
	#
	# log-file = "/home/me/enthrall.log"
	# log-file = syslog
	# log-file = stderr

	# log-level: sets the degree of verbosity in what messages are
	# sent to the log.  Must be one of the following keywords:
	# error, warn, info, verbose, debug.
	#
    log-level = debug

	# remote-shell: provides the path to the ssh binary to use.
	# If not provided, the default is simply "ssh" (whatever is
	# first in $PATH).
	#
	# remote-shell = "/alternate/path/to/ssh"

	# port: if provided, this sets the argument passed to ssh's
	# "-p" flag to set the default port to connect to on remotes.
	#
	# port = 2244

	# bind-address: if specified, the given string is passed as
	# the argument to ssh's "-b" flag.  May be useful if you have
	# multiple NICs, a v4 and a v6 address on the same interface,
	# etc.
	#
	# bind-address = "2607:f8b0:4009:803::1009"

	# identity-file: if provided, sets the argument passed to
	# ssh's "-i" flag to set the default key used to authenticate
	# to remotes.
	#
	# identity-file = "~/.ssh/enthrall-key"

	# user: if provided, this sets the default argument passed to
	# ssh's "-l" flag to set the username to connect as on the
	# remote system.
	#
	# user = "jschmoe"

	# remote-command: the default path at which to find the
	# enthrall binary on the remotes.  If not provided, the
	# default is simply whatever 'enthrall' is found first in
	# $PATH.
	#
	# remote-command = "/usr/local/bin/enthrall"

	# reconnect-max-tries: the maximum number of times to attempt
	# reconnecting to a failed remote before giving up.  Defaults
	# to 10.
	#
	# reconnect-max-tries = 5

	# reconnect-max-interval: the maximum number of seconds to
	# wait between reconnection attempts.  Default is 30.
	#
	# reconnect-max-interval = 10

	# show-focus: selects one of the following modes of providing
	# a visual hint of which node is focused (default is none):
	#
	#   dim-inactive BRIGHTNESS [DURATION NUMSTEPS]: the screens
	#   of non-active nodes are dimmed to the given fraction of
	#   their default brightness (~0.7 is probably a good starting
	#   point).  If DURATION and NUMSTEPS are specified, fade
	#   brightness transitions in NUMSTEPS steps over DURATION
	#   seconds (defaulting to 0 and 1, i.e. no fade).
	#
	#   flash-active BRIGHTNESS DURATION [NUMSTEPS]: when
	#   switching focus to a node, momentarily set its screen
	#   brightness to the given level (where "normal" is 1.0) for
	#   the specified time (in seconds).  If NUMSTEPS is
	#   specified, fade the flash out in the given number of steps
	#   over DURATION (defaulting to 1, i.e. no fade).  Note that
	#   nothing stops you from setting BRIGHTNESS to be less than
	#   1.0, so flash-active could also effectively act as a sort
	#   of "blink-active".
	#
	#   none: no visual focus hint.
	#
	# show-focus = dim-inactive 0.8 0.3 8
	# show-focus = flash-active 1.2 0.3 8
	# show-focus = flash-active 0.6 0.3 8

	# mouse-switch: selects one of the following mechanisms for
	# switching the focused node via the mouse (default is none):
	#
	#   slide: simply slide the mouse pointer off the edge of one
	#   node's display to have it slide onto the next node.
	#
	#   multi-tap COUNT WINDOW: "tap" the mouse pointer against an
	#   edge of a screen COUNT times within WINDOW seconds to
	#   switch focus in the corresponding direction.  With a COUNT
	#   of 1, WINDOW is ignored (this is equivalent to the 'slide'
	#   setting).  The maximum value for COUNT is determined by
	#   the macro EDGESTATE_HISTLEN if you really want to have to
	#   tap a lot of times to switch.
	#
	#   none: switch-by-mouse disabled.
	#
	mouse-switch = slide
	#mouse-switch = multi-tap 2 1

	# show-null-switch: selects one of the following settings for
	# determining whether or not to trigger the configured visual
	# focus-hint mechanism (see 'show-focus') when a "null" switch
	# is performed (e.g. switching left from a node that has no
	# left neighbor or switching to an already-focused node):
	#
	#   yes: do trigger visual focus hint.
	#
	#   no: do not trigger visual focus hint.
	#
	#   hotkey-only: trigger only if the null switch was performed
	#   via a hotkey, not if done via mouse-switch.
	#
	# show-null-switch: hotkey-only

	# hotkeys: hotkey[HOTKEY_STRING] = ACTION
	#
	# valid values for HOTKEY_STRING are system-dependent; some
	# possible examples on X11 might look like:
	#   "mod1+mod4+L"
	#   "mod1+mod4+control+Right"
	#
	# Similar things on OSX:
	#   "command+option+l"
	#   "command+option+control+RightArrow"
	#
	# ACTION can be one of the following:
	#
	#   focus DIRECTION (where DIRECTION is one of left, right,
	#   up, or down): switch focus to the given neighbor of the
	#   currently-focused node.
	#
	#   focus NODE: switch focus directly to the specified NODE,
	#   which can be either a string, naming a node by alias, or
	#   the keyword 'master' -- note that since master is a
	#   keyword (not a string) it should not be quoted.
	#
	#   focus previous: switch focus to the previously-focused
	#   node (toggle between the two most recently focused nodes).
	#
	#   reconnect: reset permanently-failed remotes to allow
	#   auto-reconnection to attempt to re-establish a connection.
	#
	#   quit: disconnect all remotes and terminate enthrall.

	# arrow-key-based direction-switching
	hotkey["command+option+control+LeftArrow"] = focus left
	hotkey["command+option+control+RightArrow"] = focus right
	hotkey["command+option+control+UpArrow"] = focus up
	hotkey["command+option+control+DownArrow"] = focus down

	# vi-style direction-switching
	hotkey["command+option+control+h"] = focus left
	hotkey["command+option+control+l"] = focus right
	hotkey["command+option+control+k"] = focus up
	hotkey["command+option+control+j"] = focus down

	# direct-switch keys
	hotkey["command+option+control+1"] = focus "foo"
	hotkey["command+option+control+2"] = focus master

	# focus toggle
	hotkey["command+option+control+Grave"] = focus previous

	hotkey["command+option+control+r"] = reconnect
}

# A remote block defines a remote node with a given alias.
remote "foo" {
	# hostname: if provided, sets the network hostname to connect
	# to for this remote.  If not provided, defaults to the
	# remote's alias (e.g. "foo" in this example remote block).
	#
	# hostname = "somehost"
	# hostname = "192.168.2.4"
	hostname = "10.11.189.22"

	# Each remote can also specify remote-shell, user, port,
	# bind-address, identity-file, and remote-command to provide a
	# per-remote override of the global defaults.
	#
	# remote-shell = "/alternate/path/to/ssh"
	# port = 22
	# identity-file = "~/.ssh/alternate-enthrall-key"
	# bind-address = "192.168.2.2"
	user = "srsubramanian"
	remote-command = "/home/srsubramanian/enthrall/enthrall.rhel6"

	# Platform-specific remote configuration parameters are
	# specified as strings in a special unstructured 'param' map.
	# Currently the only setting that's used for anything is
	# DISPLAY to set the environment variable of the same name on
	# X11 systems, so X11 remotes should probably have something
	# like:
	#
	param["DISPLAY"] = ":0.0"
}

# The topology block specifies how nodes are arranged relative to each
# other (i.e. who's on which side of whom).
topology {
	# Each link in the node topology is specified as:
	#
	#   NODE DIRECTION = NODE [DIRECTION]
	#
	# where NODEs are node aliases or the keyword master, as used
	# elsewhere, and DIRECTIONs are (again) one of left, right, up
	# and down.  For example:
	#
	#   master right = "foo" left
	#   "foo" right = "bar" left
	#   "fred" down = "baz" up
	#
	# This specifies a topology like this:
	#
	#    master <-> foo <-> bar
	#      ^
	#      |
	#      v
	#     baz
	#
	master right = "foo" left

	# Note, however, that the DIRECTION specifier on the right
	# side of the equal sign is optional; if omitted, this
	# specifies a one-directional link.  Example:
	#
	#   master up = "foo"
	#   "foo" down = "bar"
	#
	# If you so desire, you can use this to create exotic
	# topologies wherein (for example) a 'focus left' followed by
	# a 'focus right' does *not* leave you where you started.
	# Knock yourself out.
}
```

**Comment by zevweiss** [comment]
Hopefully fixed by commit e83eb8784971695d57d5a0f670e175da9d2b5700...

**Comment by zevweiss** [comment]
Closing; I suspect this was caused by the bug fixed in e83eb87.  Re-open if that turns out to be wrong.
